src/pokenav.c

```diff
@@ -0,0 +1,1198 @@
+    .baseTile = 0,
+};
+
+const struct WindowTemplate gUnknown_0861FA08[2] =
+{
+    {
+        .bg = 0,
+        .tilemapLeft = 1,
+        .tilemapTop = 0x16,
+        .width = 0x10,
+        .height = 0x2,
+        .paletteNum = 0,
+        .baseBlock = 0x36,
+    },
+    {
+        .bg = 0xFF,
+        .tilemapLeft = 0,
+        .tilemapTop = 0,
+        .width = 0,
+        .height = 0,
+        .paletteNum = 0,
+        .baseBlock = 0,
+    },
+};
+
+const u8 *const (MenuButtonReminders[12]) =
+{
+    gText_Navgear_ClearButtonList,
+    gText_NavgearMap_ZoomedOutButtons,
+    gText_NavgearMap_ZoomedInButtons,
+    gText_NavgearCondition_MonListButtons,
+    gText_NavgearCondition_MonStatusButtons,
+    gText_NavgearCondition_MarkingButtons,
+    gText_NavgearMatchCall_TrainerListButtons,
+    gText_NavgearMatchCall_CallMenuButtons,
+    gText_NavgearMatchCall_CheckTrainerButtons,
+    gText_NavgearRibbons_MonListButtons,
+    gText_NavgearRibbons_RibbonListButtons,
+    gText_NavgearRibbons_RibbonCheckButtons,
+};
+
+const u8 gMenuButtonReminderColor[4] =
+{
+    4, 1, 2, 0
+};
+
+const struct CompressedSpriteSheet gUnknown_0861FA4C[1] =
+{
+    {
+        .data = gUnknown_0861F5B0,
+        .size = 0x1000,
+        .tag = 0,
+    }
+};
+
+const struct SpritePalette gUnknown_0861FA54[2] =
+{
+    {
+        .data = gUnknown_0861F590,
+        .tag = 0,
+    },
+    {
+        .data = NULL,
+        .tag = 0,
+    }
+};
+
+const struct CompressedSpriteSheet gUnknown_0861FA64 =
+{
+    .data = gPokenavLeftHeaderHoennMap_Gfx,
+    .size = 0xC00,
+    .tag = 2
+};
+
+extern struct UnknownStruct_0203CF40 *gUnknown_0203CF40;
+extern u8 gUnknown_0203CF3C;
+extern const struct SpriteTemplate gUnknown_0861FB04;
+
+u32 sub_81C7078(u32 (*a0)(s32), u32 a1)
+{
+    u16 taskId;
+
+    if (!is_c1_link_related_active())
+        taskId = CreateTask(sub_81C7170, a1);
+    else
+        taskId = CreateTask(sub_81C71E4, a1);
+
+    SetWordTaskArg(taskId, 1, (u32)a0);
+
+    gTasks[taskId].data[3] = gUnknown_0203CF3C;
+    return ((gUnknown_0203CF3C++) << 16) | taskId;
+}
+
+bool32 sub_81C70D8(u32 a0)
+{
+    u32 v1 = a0 & 0xFFFF;
+    u32 v2 = a0 >> 16;
+    if (gTasks[v1].isActive
+        && (gTasks[v1].func == sub_81C7170 || gTasks[v1].func == sub_81C71E4)
+        && gTasks[v1].data[3] == v2)
+    {
+    
+        return TRUE;
+    }
+    else
+        return FALSE;
+}
+
+bool32 sub_81C7124(u32 a0)
+{
+    s32 i;
+    for (i = 0; i < 16; i++)
+    {
+        if (gTasks[i].isActive
+            && (gTasks[i].func == sub_81C7170 || gTasks[i].func == sub_81C71E4))
+        {
+            u32 arg = GetWordTaskArg((u8)i, 1);
+            if (arg == a0)
+                return TRUE;
+        }
+    }
+    return FALSE;
+}
+
+void sub_81C7170(u8 taskId)
+{
+    s16 *dataPtr;
+    u32 (*func)(u32);
+    bool32 exitLoop;
+
+    func = (u32 (*)(u32))GetWordTaskArg(taskId, 1);
+    dataPtr = gTasks[taskId].data;
+    exitLoop = FALSE;
+    while (!exitLoop)
```

Comment:
Is this temp var really needed here?

Reply:
Yes, unfortunately. The ASM saves the exitLoop value in a separate register as though it could change during the loop.